linera-execution/src/lib.rs

```diff
@@ -334,6 +336,25 @@ pub trait BaseRuntime {
     /// Reads the balance of the chain.
     fn read_chain_balance(&mut self) -> Result<Amount, ExecutionError>;
 
+    /// Reads the owner balance.
+    fn read_owner_balance(&mut self, owner: Owner) -> Result<Amount, ExecutionError>;
+
+    /// Transfers amount from source to destination
+    fn transfer(
+        &mut self,
+        source: Option<Owner>,
+        destination: Account,
+        amount: Amount,
+    ) -> Result<(), ExecutionError>;
+
+    /// Claims amount from source to destination
+    fn claim(
+        &mut self,
+        source: Account,
+        destination: Account,
+        amount: Amount,
+    ) -> Result<(), ExecutionError>;
```

Comment:
Do `transfer` and `claim` really belong in `BaseRuntime`? They can't be called by a service, so I imagine they should be in `ContractRuntime`.